src/features/world/roomMachine.ts

```diff
@@ -89,63 +88,76 @@ export type MachineInterpreter = Interpreter<
   ChatContext,
   any,
   RoomEvent,
-  RoomState
+  any,
+  any
 >;
 
 /**
  * Machine which handles room events
  */
 export const roomMachine = createMachine<ChatContext, RoomEvent, RoomState>({
-  initial: "idle",
+  initial: "initialising",
```

Comment:
I've updated the state machine so that initialisation only sets up a single client to the backend. 

Subsequent calls to change room will not reinitialize another client.